### PR TITLE
lucenepp: 3.0.7 -> 3.0.8

### PR DESCRIPTION
**Pull request by nagy**
###### Motivation for this change

The motivation is mostly to get rid of `boost155` eventually.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
  - [x] NixOS
  - [ ] macOS
  - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134481` at 6c0d2861 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c1e8f7366ca7d6074aaef7d71c38454b">poedit</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>lucenepp</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/lucene++/default.nix:27:3:
```
   |
27 |   meta = {
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by davidak** [comment]
@nagy could anything be done to fix build on aarch64-linux or should it be disabled?

**Comment by nagy** [comment]
It looks like it is just the test phase. Maybe we can skip the tests on aarch64-linux. I am still working on the `poedit` fails.

**Comment by nagy** [comment]
I could fix the `poedit` build failures. Lets see if the aarch64-linux tests still fail.

File: pkgs/development/libraries/lucene++/default.nix [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
+            --replace "add_subdirectory(gtest)" ""
   '';
 
-  cmakeFlags = [ "-DGTEST_INCLUDE_DIR=${gtest}/include" ];
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost gtest ];
+  checkPhase= ''
+    runHook preCheck
+    LD_LIBRARY_PATH=$PWD/src/contrib:$PWD/src/core \
+            src/test/lucene++-tester
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lucene++/default.nix:27]: Did they remove the test target? Maybe we can set LD_LIBRARY_PATH in preCheck?

Comment (nagy) [reply pkgs/development/libraries/lucene++/default.nix:27]: > Did they remove the test target? 

Yes, at least this particular target went away and I couldn't find another one. According to the README they want you to run this binary to run the tests. https://github.com/luceneplusplus/LucenePlusPlus/blob/rel_3.0.8/README.md#to-run-unit-test-suite

> Maybe we can set LD_LIBRARY_PATH in preCheck?

You mean like this ? :
```nix
preCheck = ''
  export LD_LIBRARY_PATH=$PWD/src/contrib:$PWD/src/core
'';
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 134481` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>lucenepp</li>
<li>poedit</li>
</ul>
</details>

File: pkgs/development/libraries/lucene++/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/lucene++/default.nix:27]: > You mean like this ? :

Yeah but that is only useful when using test target.

Comment (nagy) [reply pkgs/development/libraries/lucene++/default.nix:27]: What was the last sentiment here? Do you want me to change it? I dont see much use in that.

File: pkgs/development/libraries/lucene++/default.nix [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
-           -e '/include.*CMakeExternal/d' \
-           CMakeLists.txt
+     substituteInPlace src/test/CMakeLists.txt \
+            --replace "add_subdirectory(gtest)" ""
   '';
 
-  cmakeFlags = [ "-DGTEST_INCLUDE_DIR=${gtest}/include" ];
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost gtest ];
+  checkPhase= ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lucene++/default.nix:24]:
```suggestion
  checkPhase = ''
```

**Pull request merged**